### PR TITLE
Data, Flink, Spark: Test deletes with format-version=3

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/TestMetadataTableFilters.java [hunk 1]
```diff
@@ -441,12 +441,6 @@ public void testPartitionSpecEvolutionRemovalV2() {
     assertThat(tasks).hasSize(expectedScanTaskCount(3));
   }
 
-  private DeleteFile posDelete(Table table, DataFile dataFile) {
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/TestMetadataTableFilters.java:444]: this addresses https://github.com/apache/iceberg/pull/11485#discussion_r1840262214

File: data/src/test/java/org/apache/iceberg/data/DeleteReadTests.java [hunk 2]
```diff
@@ -82,12 +83,15 @@ public abstract class DeleteReadTests {
...
+
   @Parameters(name = "fileFormat = {0}")
   public static Object[][] parameters() {
     return new Object[][] {
-      new Object[] {FileFormat.PARQUET},
-      new Object[] {FileFormat.AVRO},
-      new Object[] {FileFormat.ORC}
+      new Object[] {FileFormat.PARQUET, 2},
+      new Object[] {FileFormat.AVRO, 2},
+      new Object[] {FileFormat.ORC, 2}
```
Comment (amogh-jahagirdar) [inline data/src/test/java/org/apache/iceberg/data/DeleteReadTests.java:94]: Same as below, any reason we shouldn't just add some V3 cases here?

**Review by amogh-jahagirdar — COMMENTED** [review]
Minor naming comment

File: mr/src/test/java/org/apache/iceberg/mr/TestInputFormatReaderDeletes.java [hunk 3]
```diff
@@ -49,18 +49,18 @@ public class TestInputFormatReaderDeletes extends DeleteReadTests {
...
-      {FileFormat.ORC, "IcebergInputFormat"},
-      {FileFormat.PARQUET, "MapredIcebergInputFormat"},
-      {FileFormat.AVRO, "MapredIcebergInputFormat"},
-      {FileFormat.ORC, "MapredIcebergInputFormat"},
+      {FileFormat.PARQUET, 2, "IcebergInputFormat"},
+      {FileFormat.AVRO, 2, "IcebergInputFormat"},
+      {FileFormat.ORC, 2, "IcebergInputFormat"},
+      {FileFormat.PARQUET, 2, "MapredIcebergInputFormat"},
+      {FileFormat.AVRO, 2, "MapredIcebergInputFormat"},
+      {FileFormat.ORC, 2, "MapredIcebergInputFormat"},
```
[the remark this answers is not in the document]
Comment (nastra) [reply mr/src/test/java/org/apache/iceberg/mr/TestInputFormatReaderDeletes.java:63]: yeah that probably makes sense to test these with V3 now too

File: spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java [hunk 4]
```diff
@@ -99,16 +100,21 @@ public class TestSparkReaderDeletes extends DeleteReadTests {
...
 
-  @Parameter(index = 1)
+  @Parameter(index = 2)
   private boolean vectorized;
 
-  @Parameters(name = "format = {0}, vectorized = {1}")
+  @Parameter(index = 3)
+  private PlanningMode planningMode;
+
+  @Parameters(name = "format = {0}, formatVersion = {1}, vectorized = {2}, planningMode = {3}")
```
[the remark this answers is not in the document]
Comment (nastra) [reply spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java:109]: yep makes sense, updated

File: data/src/test/java/org/apache/iceberg/data/DeleteReadTests.java (hunk 2 above)
Comment (nastra) [reply data/src/test/java/org/apache/iceberg/data/DeleteReadTests.java]: yep we should, updated

**Pull request merged**